src/custom-element.js

```diff
@@ -123,92 +123,103 @@ export function stubElements(win) {
  * @param {!AmpElement} element
  */
 export function applyLayout_(element) {
-  let widthAttr = element.getAttribute('width');
-  let heightAttr = element.getAttribute('height');
-  const sizesAttr = element.getAttribute('sizes');
   const layoutAttr = element.getAttribute('layout');
+  const widthAttr = element.getAttribute('width');
+  const heightAttr = element.getAttribute('height');
+  const sizesAttr = element.getAttribute('sizes');
+
+  // Input layout attributes.
+  const inputLayout = layoutAttr ? parseLayout(layoutAttr) : null;
+  assert(inputLayout || !layoutAttr, 'Unknown layout: %s', layoutAttr);
```

Comment:
Is it cool to switch to assert? Sorry I don't know the practical difference in Javascript between that and exceptions ... I guess I wonder whether exceptions might be caught by something vs. assertions give up on the entire document, but I'm not familiar with assert in Javascript. Could be a non-issue, sorry.